Frends.FTP.ListFiles/Frends.FTP.ListFiles/ListFiles.cs

```diff
@@ -0,0 +1,105 @@
+﻿using Frends.FTP.ListFiles.Definitions;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentFTP;
+using System.Text.RegularExpressions;
+
+namespace Frends.FTP.ListFiles
+{
+    ///<summary>
+    /// Files task.
+    /// </summary>
+    public class FTP
+    {
+        /// <summary>
+        /// Gets the list of files from the FTP(S) source.
+        /// [Documentation](https://tasks.frends.com/tasks/frends-tasks/Frends.FTP.ListFiles)
+        /// </summary>
+        /// <param name="input">Input parameters</param>
+        /// <param name="cancellationToken">Token to stop task. This is generated by Frends.</param>
+        /// <returns>List { DateTime Modified, string Name, string FullPath, long Size, bool Success }</returns>
+        public static async Task<Result> ListFiles([PropertyTab] Input input, CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrWhiteSpace(input.Host)) throw new Exception("Host required.");
+            if (!string.IsNullOrWhiteSpace(input.User) && string.IsNullOrWhiteSpace(input.Password) || string.IsNullOrWhiteSpace(input.User) && !string.IsNullOrWhiteSpace(input.Password)) throw new Exception("Credentials required.");
```

Comment:
Can we split this if into multiple ones (at the OR part) since this is way too confusing to comprehend as a reader.